ServiceFinder/ServiceFinder.BLL/Services/AssistanceCategoryService.cs

```diff
@@ -3,13 +3,62 @@
 using ServiceFinder.BLL.Models;
 using ServiceFinder.DAL.Entites;
 using ServiceFinder.DAL.Interfaces;
+using ServiceFinder.DAL.PaginationObjects;
 
 namespace ServiceFinder.BLL.Services
 {
     public class AssistanceCategoryService : GenericService<AssistanceCategoryEntity, AssistanceCategory>, IAssistanceCategoryService
     {
-        public AssistanceCategoryService(IAssistanceCategoryRepository serviceCategoryRepository, IMapper mapper) : base(serviceCategoryRepository, mapper)
+        private readonly IUserProfileRepository _userProfileRepository;
+        private readonly IAssistanceCategoryRepository _assistanceCategoryRepository;
```

Comment:
Remove this unread private field '_assistanceCategoryRepository' or refactor the code to use its value. (https://rules.sonarsource.com/csharp/RSPEC-4487)